src/main/kotlin/io/github/moulberry/notenoughupdates/guifeatures/SkyMallDisplay.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2023 NotEnoughUpdates contributors
+ *
+ * This file is part of NotEnoughUpdates.
+ *
+ * NotEnoughUpdates is free software: you can redistribute it
+ * and/or modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * NotEnoughUpdates is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with NotEnoughUpdates. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.moulberry.notenoughupdates.guifeatures
+
+import io.github.moulberry.notenoughupdates.NotEnoughUpdates
+import io.github.moulberry.notenoughupdates.autosubscribe.NEUAutoSubscribe
+import io.github.moulberry.notenoughupdates.util.SBInfo
+import io.github.moulberry.notenoughupdates.util.SkyBlockTime
+import io.github.moulberry.notenoughupdates.util.Utils
+import net.minecraftforge.client.event.ClientChatReceivedEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.util.regex.Pattern
+
+@NEUAutoSubscribe
+class SkyMallDisplay {
+
+    private val pattern = Pattern.compile("§r§eNew buff§r§r§r: (.*)§r")
+
+    @SubscribeEvent(receiveCanceled = true)
+    fun onChatReceive(event: ClientChatReceivedEvent) {
+        if (!NotEnoughUpdates.INSTANCE.hasSkyblockScoreboard()) return
+        if (SBInfo.getInstance().getLocation() != "mining_3") return
+
+        val matcher = pattern.matcher(event.message.formattedText)
+        if (!matcher.matches()) return
+
+        matcher.group(1).let { message ->
+            SkyMallVariant.values().find { it.chatMessage == message }
+        }.also {
+            if (it != null) {
+                currentVariant = it
+                println("found ${it.displayName}")
+                update()
+            } else {
+                println("nothing found!")
+            }
+        }
```

Comment:
```suggestion
        val message = matcher.group(1) ?: return
        currentVariant = SkyMallVariant.values().find { it.chatMessage == message }
        update()
```

Reply:
yeah i dont need the debugs anymore ur right

Reply:
Implemented.